Last batch of Unit Tests

MatonAnthony opened the pull request.

MatonAnthony committed:
Test exampleColonHeading linting

MatonAnthony committed:
Test fontElement linting

MatonAnthony committed:
Test HTMLComments linting

MatonAnthony committed:
Test HTTPLinks linting

MatonAnthony committed:
Test incorrectlyWrappedSidebarMacro linting

MatonAnthony committed:
Test invalidMacros linting

MatonAnthony committed:
Test lineLength linting

MatonAnthony committed:
Test macroSyntaxError linting

MatonAnthony committed:
Test nameAttribute linting

MatonAnthony committed:
Test oldUrls linting

MatonAnthony committed:
Test preWithoutClass linting

MatonAnthony committed:
Test shellPrompts

MatonAnthony committed:
Test styleAttribute linting

MatonAnthony committed:
Test summaryHeading linting

MatonAnthony committed:
Test unnecessaryMacroParams linting

MatonAnthony committed:
Test urlInLinkTitle linting

MatonAnthony committed:
Test wrongHighlightedLine linting

MatonAnthony committed:
Test wrongSyntaxClass linting

MatonAnthony committed:
Improve testrunner

MatonAnthony committed:
Fix errors made during refactoring

MatonAnthony committed:
Enable the running tests

Elchi3 reviewed (approved):
Let's land this second batch of rules.
I would have put them in a folder called "rules" instead of "lib".

MatonAnthony commented:
I'm open to change it, I just followed what I saw in other npm packages such as https://github.com/mikedeboer/node-github

Elchi3 commented:
I think when talking about lints and tests, the terms create ambiguity. So, one could ask where the actual mdn lints are. In the test folder? no, there we test the tests. I think "rules" would make that more clear, but we can change this in a follow-up if we want to.

MatonAnthony committed:
Remove deadcode

Remove testutils.js as it was used to run the tests on the old SDK based add-on
